File: MediaLoader/src/main/java/com/app/medialoader/data/file/BaseFileDataSource.java
package com.app.medialoader.data.file;


import com.app.medialoader.data.file.cleanup.DiskLruCache;
import com.app.medialoader.utils.FileUtil;
import com.app.medialoader.utils.LogUtil;

import java.io.File;
import java.io.IOException;

/**
 * {@link FileDataSource}的通用实现
 *
 * @author vincanyang
 */
public abstract class BaseFileDataSource implements FileDataSource {

    protected File mOriginFile;

    protected DiskLruCache mDiskLruStorage;

    public BaseFileDataSource(File file, DiskLruCache diskLruStorage) {
        mOriginFile = file;
        mDiskLruStorage = diskLruStorage;
        try {
            FileUtil.mkdirs(file.getParentFile());
        } catch (IOException e) {
            LogUtil.e(e);
        }
    }

    @Override
    public void close() throws IOException {
        mDiskLruStorage.save("", mOriginFile);
    }

}

File: MediaLoader/src/main/java/com/app/medialoader/data/file/BaseFileDataSource.kt
package com.app.medialoader.data.file

import com.app.medialoader.data.file.cleanup.DiskLruCache
import com.app.medialoader.utils.FileUtil.mkdirs
import com.app.medialoader.utils.LogUtil.e
import java.io.File
import java.io.IOException

/**
 * [FileDataSource]的通用实现
 *
 * @author vincanyang
 */
abstract class BaseFileDataSource(
    protected var mOriginFile: File,
    private var mDiskLruStorage: DiskLruCache?
) : FileDataSource {
    @Throws(IOException::class)
    override fun close() {
        mDiskLruStorage?.save("", mOriginFile)
    }

    init {
        try {
            mkdirs(mOriginFile.parentFile)
        } catch (e: IOException) {
            e(e)
        }
    }
}